SConstruct

```diff
@@ -430,6 +415,18 @@ env.modules_detected = modules_detected
 opts.Update(env, {**ARGUMENTS, **env.Dictionary()})
 Help(opts.GenerateHelpText(env))
 
+
+# FIXME: Tool assignment happening at this stage is a direct consequence of getting the platform logic AFTER the SCons
+# environment was already been constructed. Fixing this would require a broader refactor where all options are setup
+# ahead of time with native validator/converter functions.
+custom_tools = ["default"]
+try:  # Platform custom tools are optional
+    custom_tools = detect.get_tools(env)
+except AttributeError:
+    pass
+env = env.Clone(tools=custom_tools)
```

Comment:
This fixes the problem of requiring an `env` to parse options, but requiring options to be parsed to register the env with the right tools - at least in my early testing only for Windows mingw builds on Linux.

I haven't read much about what `Environment.Clone()` actually does so we need to do more research to confirm that it's the right tool for the job.

Reply:
This shouldn't be necessary if using the above empty tool argument.
```suggestion
for tool in custom_tools:
	env.Tool(tool)
```

Reply:
Interesting! Actually I think the `Clone()` wasn't required either way, it also seems to work fine with the original `Environment(tools=["default"])`.

The issue in https://github.com/godotengine/godot/pull/101715/files#r1920844908 was due to `mingw` not being properly set as custom tool when just doing `scons p=windows` on Linux, because our logic for that was incomplete. Why it worked before our PRs though, I'm not sure.

Reply:
> The issue in https://github.com/godotengine/godot/pull/101715/files#r1920844908 was due to `mingw` not being properly set as custom tool when just doing `scons p=windows` on Linux, because our logic for that was incomplete. Why it worked before our PRs though, I'm not sure.

Actually I think the reason it failed in #101715 is because of the removal of `default`.

So I'm still unsure whether it's a good idea to do `tools=[]` and then register some custom tools, or `tools=["default"]` to have sane defaults and then add custom tools (like we currently do in `master`).

Reply:
> Actually I think the reason it failed in #101715 is because of the removal of `default` here:

Or not, there must be something more linked with the `Environment()` constructor.